### PR TITLE
Issue #9558: update example of AST for TokenTypes.RBRACK

### PR DESCRIPTION
**Pull request by SGanguly1999**
Closes: #9558

image

`$ cat Demo.java`

```
class Demo
{
    public static void main(String args[])
    {
        int a[];
    }
} 
```
`$ java -jar checkstyle-8.41-all.jar -t Demo.java
`
```
CLASS_DEF -> CLASS_DEF [2:0]
|--MODIFIERS -> MODIFIERS [2:0]
|--LITERAL_CLASS -> class [2:0]
|--IDENT -> Demo [2:6]
`--OBJBLOCK -> OBJBLOCK [3:0]
    |--LCURLY -> { [3:0]
    |--METHOD_DEF -> METHOD_DEF [4:4]
    |   |--MODIFIERS -> MODIFIERS [4:4]
    |   |   |--LITERAL_PUBLIC -> public [4:4]
    |   |   `--LITERAL_STATIC -> static [4:11]
    |   |--TYPE -> TYPE [4:18]
    |   |   `--LITERAL_VOID -> void [4:18]
    |   |--IDENT -> main [4:23]
    |   |--LPAREN -> ( [4:27]
    |   |--PARAMETERS -> PARAMETERS [4:39]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [4:39]
    |   |       |--MODIFIERS -> MODIFIERS [4:39]
    |   |       |--TYPE -> TYPE [4:39]
    |   |       |   `--ARRAY_DECLARATOR -> [ [4:39]
    |   |       |       |--IDENT -> String [4:28]
    |   |       |       `--RBRACK -> ] [4:40]
    |   |       `--IDENT -> args [4:35]
    |   |--RPAREN -> ) [4:41]
    |   `--SLIST -> { [5:4]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [6:13]
    |       |   |--MODIFIERS -> MODIFIERS [6:13]
    |       |   |--TYPE -> TYPE [6:13]
    |       |   |   `--ARRAY_DECLARATOR -> [ [6:13]
    |       |   |       |--LITERAL_INT -> int [6:8]
    |       |   |       `--RBRACK -> ] [6:14]
    |       |   `--IDENT -> a [6:12]
    |       |--SEMI -> ; [6:15]
    |       `--RCURLY -> } [7:4]
    `--RCURLY -> } [8:0]
```

Required part:
```
|--VARIABLE_DEF -> VARIABLE_DEF [6:13]
|      |--MODIFIERS -> MODIFIERS [6:13]
|      |--TYPE -> TYPE [6:13]
|      |      `--ARRAY_DECLARATOR -> [ [6:13]
|      |             |--LITERAL_INT -> int [6:8]
|      |             `--RBRACK -> ] [6:14]
|      `--IDENT -> a [6:12]
|--SEMI -> ; [6:15]
```

Expected :

```
|--VARIABLE_DEF -&gt; VARIABLE_DEF
|      |--MODIFIERS -&gt; MODIFIERS
|      |--TYPE -&gt; TYPE
|      |   `--ARRAY_DECLARATOR -&gt; [
|      |       |--LITERAL_INT -&gt; int
|      |       `--RBRACK -&gt; ]
|      `--IDENT -&gt; a
|--SEMI -&gt; ;

```

### CODE REVIEW

**Comment by SGanguly1999** [comment]
Members please review this when you are free

**Comment by SGanguly1999** [comment]
@timurt could you please review this PR when you are free

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1244,6 +1244,19 @@
     /**
      * The {@code ]} symbol.
      *
+     * <p>For example:</p>
+     * <pre>
+     * int a[];
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * |--TYPE -&gt; TYPE
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1253]: hell, it seems like it should start with `VARIABLE_DEF`, please update

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (SGanguly1999) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @timurt done

Comment (timurt) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: sorry, i just noticed that I wrote 'hell' instead of 'hello' :|

**Review by timurt — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (SGanguly1999) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: No no its okay...XD

**Pull request merged**